erizo/src/erizo/media/ExternalInput.cpp

```diff
@@ -126,9 +126,9 @@ int ExternalInput::init() {
     op_->init(om, this);
   } else {
     needTranscoding_ = true;
-    inCodec_.initDecoder(st->codec);
+    inCodec_.initDecoder(st->codecpar);
 
-    bufflen_ = st->codec->width*st->codec->height*3/2;
+    bufflen_ = inCodec_.codec_context_->width*inCodec_.codec_context_->height*3/2;
```

Comment:
Well, ExternalInput is a particular thing, since I've started there before realize the common API (Coder, CoderCodec, etc.) should be used, then I've updated it but not deeply read and refactor the code, so consider it pending for now please. I'll post bellow a list of the things I am considering that needs to be finished.